IdentityCore/src/webview/MSIDWebviewSession.m

```diff
@@ -26,34 +26,77 @@
 //------------------------------------------------------------------------------
 
 #import "MSIDWebviewSession.h"
+#import "MSIDWebviewInteracting.h"
+#import "MSIDNotifications.h"
 
 @implementation MSIDWebviewSession
 
 - (instancetype)initWithWebviewController:(NSObject<MSIDWebviewInteracting> *)webviewController
                                   factory:(MSIDWebviewFactory *)factory
+                              redirectUri:(NSString *)redirectUri
                              requestState:(NSString *)state
-{
-    return [self initWithWebviewController:webviewController
-                                   factory:factory
-                              requestState:state
-                               verifyState:NO];
-}
 
-
-- (instancetype)initWithWebviewController:(NSObject<MSIDWebviewInteracting> *)webviewController
-                                  factory:(MSIDWebviewFactory *)factory
-                             requestState:(NSString *)state
-                              verifyState:(BOOL)verifyState
 {
     self = [super init];
     if (self)
     {
         _webviewController = webviewController;
+        _webviewController.webviewNotifiableDelegate = self;
         _factory = factory;
         _requestState = state;
-        _verifyState = verifyState;
+        _redirectUri = redirectUri;
     }
+    
     return self;
 }
 
+#pragma mark - MSIDWebviewDelegate
+- (void)webAuthDidStartLoad:(NSURL *)url
```

Comment:
Does webviewsession need to be notifiable delegate? I think there's nothing that's necessary from the webviewsession to fire the notification.

Reply:
nope moved to MSIDNotifications class methods